src/util/bip32.rs

```diff
@@ -337,33 +338,36 @@ impl ExtendedPubKey {
     }
 }
 
-impl ToBase58 for ExtendedPrivKey {
-    fn base58_layout(&self) -> Vec<u8> { 
-        let mut ret = Vec::with_capacity(78);
-        ret.extend(match self.network {
-            Network::Bitcoin => [0x04, 0x88, 0xAD, 0xE4],
-            Network::Testnet => [0x04, 0x35, 0x83, 0x94]
-        }.iter().cloned());
-        ret.push(self.depth as u8);
-        ret.extend(self.parent_fingerprint[..].iter().cloned());
+impl ToString for ExtendedPrivKey {
+    fn to_string(&self) -> String {
+        let mut ret = [0; 78];
+        ret[0..4].copy_from_slice(match self.network {
+            Network::Bitcoin => &[0x04, 0x88, 0xAD, 0xE4][..],
```

Comment:
Would be nice to break these version bytes (and other constants elsewhere) into an external repo. Maybe not now, but soon?

Reply:
Does that really need to be an external repo? We could move the address parsing stuff into its own repo and expose these constants but having a repo for constants collection just seems like naval-gazing.

Reply:
Agreed. @TheBlueMatt thoughts? This would add a formal dependency to basically every Bitcoin lib but it would eliminate a bunch of constant-conversion code.

Reply:
As it stands, there are some `hrp` constants defined in `rust-bech32-bitcoin` for multiple chains. One approach is to move those constants into this library's `constants` module, but that creates a circular dependency.

As written, this code works for Bitcoin only, and any additional chain support requires updating many places within the code.

Not that I want to support lots of altcoins, but the lightning effort gets very interesting in the cross-chain channels case.

The simplest way forward is including only the constants embedded in nice `enum`s, without classification code.